homework/index.js

```diff
@@ -0,0 +1,24 @@
+const inputValue = document.querySelector('#search');
+const searchButton = document.querySelector('.searchButton');
+const reposContainer = document.querySelector('.main__profile-repos');
+// const clientId = '1ebd35cdc9277ed1ee8c';
+// const clientSecret = '0ecdc67b56e4ca7c3b4710b60f135d52a8f11a32';
+const fetchRepo = async () => {
+  // const apiCall = await fetch(`https://api.github.com/users/${user}?clientId=${clientId}&clientSecret=${clientSecret}`);
+  const apiCall = await fetch('https://api.github.com/HackYourFuture/repos?callback=?');
+
+  const data = await apiCall.json();
+  return {
+    data,
+  };
+};
+
+const showData = () => {
+  fetchRepo(inputValue.value).then((res) => {
+    reposContainer.innerHTML = `Repos:<span class = "main_profile">${res.data.public_repos}</span>`;
```

Comment:
Ok, here you need a bit more feedback I think. This is the key part of the exercise: how to translate the repositories to entries in a select list.

First, I don't know where you go the `public_repos` from? That's not an existing property on the data: hence you get `undefined`.

If you remove that again (and fix the endpoint url), you would see that your application displays this:

> Repos:[object Object],[object Object],[object Object],... (etc)

That's because the 'repositories' data is converted into a string. Converting an array to a string will work: it will list all elements in the array, separated by a comma. And because the elements in this array are 'repository' *objects*, it will list them as `object Object`. We want to display just the *name* of these repositories. We can do that by *mapping* each repository item in the repositories array into a string containing just the repository name! Like this:

```
const repositoryNames = repositories.map(r => r.name);
```

We can then use those to display them in the console (`console.log(repositoryNames)`), append them as text to your container, or, ultimately: use them to populate your `<select>` list.

Could you give that a try?